enterprise/ha/src/test/java/org/neo4j/ha/UpdatePullerSwitchIT.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ha;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.neo4j.cluster.ClusterSettings;
+import org.neo4j.cluster.InstanceId;
+import org.neo4j.graphdb.DynamicLabel;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.helpers.collection.Iterables;
+import org.neo4j.kernel.ha.HaSettings;
+import org.neo4j.kernel.ha.HighlyAvailableGraphDatabase;
+import org.neo4j.kernel.ha.SlaveUpdatePuller;
+import org.neo4j.kernel.ha.UpdatePuller;
+import org.neo4j.test.ha.ClusterManager;
+import org.neo4j.test.ha.ClusterRule;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.neo4j.kernel.ha.HaSettings.tx_push_factor;
+
+public class UpdatePullerSwitchIT
+{
+    @Rule
+    public final ClusterRule clusterRule = new ClusterRule( getClass() );
+    private ClusterManager.ManagedCluster managedCluster;
+
+    @Before
+    public void setup() throws Exception
+    {
+        managedCluster = clusterRule.provider( ClusterManager.clusterOfSize( 2 ) )
+                                    .config( tx_push_factor, "0" )
+                                    .config( HaSettings.pull_interval, "100s" )
+                                    .startCluster();
+        managedCluster.await( ClusterManager.masterAvailable() );
```

Comment:
Is this check needed? I think there is already a default `ClusterManager#allSeesAllAsAvailable()` check in `ClusterRule#startCluster()`.